scroll-animations-1/Overview.bs

```diff
@@ -602,3 +653,146 @@ spec: cssom-view-1; type: dfn;
+	are both [=shorthand property|shorthands=]
+	that set 'animation-delay-start' and 'animation-delay-end'
+	together in a single declaration.
+	If the <<'animation-delay-end'>> value is omitted,
+	it is set to zero.
+	If a <<timeline-range-name>> alone is specified,
+	'animation-delay-start' is set to that name plus ''0%''
+	and 'animation-delay-end' is set to that name plus ''100%''.
+
+	<pre class="propdef">
+		Name: animation-delay-start
+		Value: [ <<time>> | <<timeline-range-name>> <<percentage>>
+		Initial: 0s
+		Applies to: all elements
+		Inherited: no
+		Percentages: relative to the specified [=named timeline range=]
+		Computed value: list, each item either a duration or a timeline range and progress percentage
+		Animatable: no
+	</pre>
+
+	Defines a delay between the start of the animation
+	and when it begins executing
+	(i.e. when keyframes attached to 0% progress are mapped).
+
+	<dl dfn-for="animation-delay-start, animation-delay-end" dfn-type=value>
+		<dt><dfn><<time>></dfn>
+		<dd>
+			Defines the delay as a duration. See [[CSS-ANIMATIONS-1]].
+		<dt><dfn><<timeline-range-name>> <<percentage>></dfn>
+		<dd>
+			Calculates the delay as the difference between
+			the start of the animation
+			and the specified point on the [=timeline=].
+	</dl>
+
+	<pre class="propdef">
+		Name: animation-delay-end
+		Value: [ <<time>> | <<timeline-range-name>> <<percentage>>
+		Initial: 0s
+		Applies to: all elements
+		Inherited: no
+		Percentages: relative to the specified [=named timeline range=]
+		Computed value: list, each item either a duration or a timeline range and progress percentage
+		Animatable: no
+	</pre>
+
+	Defines a delay between the end of the animation
+	and when it ends executing
+	(i.e. when keyframes attached to 100% progress are mapped).
+	Values are as for 'animation-delay-start',
+	but calculating backwards from the end,
+	i.e. a ''2s'' delay effectively shortens the animation duration,
+	whereas ''-2s'' lengthens it.
+
+## Reporting Timeline Range Progress: the ''getTime()'' method ## {#named-range-getTime}
+
+	Progress through named ranges is exposed on the {{AnimationTimeline}} object
+	by the {{AnimationTimeline/getCurrentTime()}} method:
+
+	<pre class="webidl">
+		[Exposed=Window]
+		interface AnimationTimeline {
+		    readonly attribute double? currentTime;
+		    CSSNumericValue getCurrentTime(CSSOMString? rangeName);
+		};
+	</pre>
+
+	<dl class=methods dfn-type=method dfn-for="AnimationTimeline">
+		<dt><dfn>getCurrentTime(rangeName)</dfn>
+		<dd>
+			If a rangeName is provided,
+			and is a valid [=named timeline range=] on this {{AnimationTimeline}},
+			returns the percentage progress through that range.
+
+			If a rangeName is provided,
+			and is not a valid [=named timeline range=] on this {{AnimationTimeline}},
+			returns null.
+
+			If a rangeName is not provided,
+			returns {{AnimationTimeline/currentTime}},
+			representing any time-based value using milliseconds (''ms'').
+	</dl>
```

Comment:
Rewriting to the preferred algorithmic form, and using the predefined variables from IDL:

```suggestion
	<div algorithm="AnimationTimeline.getCurrentTime()">
		The <code><dfn method for=AnimationTimeline>getCurrentTime(optional |rangeName|)</dfn></code> [=method steps=] are:

		1. If [=this=] is an [=inactive timeline=],
			return [=null=].

		1. If |rangeName| is provided:

			1. If |rangeName| is a valid [=named timeline range=] on [=this=],
				let |progress| be the current progress through that range,
				expressed as a percentage value between 0 and 100.

				Create a [=new unit value=] from (|progress|, "percent")
				and return it.

			2. Otherwise, return [=null=].

		2. Let |current time| 
			be the value of [=this's=] {{AnimationTimeline/currentTime}} internal slot.

			Create a [=new unit value=] from (|current time|, "ms")
			and return it.
	</div>
```